### PR TITLE
Adding auth templates for iOS and Android

### PR DESCRIPTION
**Pull request by bhariharan**
This PR adds auth templates for `iOS` and `Android` so that they can be used with `forceios` and `forcedroid` to create sample apps.

### CODE REVIEW

**Comment by bhariharan** [comment]
The `iOS` piece is good to go -> updated to the latest specs from the `Templates` repo, updated it to use the new `SalesforceSwiftSDK`, specified pod versions for all our dependencies and some other cleanup -> it compiles and runs fine. `Android` should be ready for review too - there are a couple of minor bugs to fix there, but I can make them in a separate PR.

File: iOSIDPTemplate/Authenticator/Utilities.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ */
+import Foundation
+import UIKit
+import SalesforceSDKCore
+
+let app_one_key = "app-one"
+
+let app_two_key = "app-two"
+
+let app_one_default_url = "smartsyncexplorer"
```
Comment (wmathurin) [inline iOSIDPTemplate/Authenticator/Utilities.swift:32]: Do we want that in the template?

File: iOSIDPTemplate/Images.xcassets/sam.imageset/Contents.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "idiom" : "universal",
+      "filename" : "User2_2.png",
```
Comment (wmathurin) [inline iOSIDPTemplate/Images.xcassets/sam.imageset/Contents.json:5]: Does the app show fake users?

File: iOSIDPTemplate/Authenticator/Utilities.swift (hunk 1 above)
Comment (bhariharan) [reply iOSIDPTemplate/Authenticator/Utilities.swift:32]: I think so. By default, the template references our sample apps - `SmartSyncExplorer` and `AccountEditor`.

File: iOSIDPTemplate/Images.xcassets/sam.imageset/Contents.json (hunk 2 above)
Comment (bhariharan) [reply iOSIDPTemplate/Images.xcassets/sam.imageset/Contents.json:5]: No - the users are real. The icons are default avatars in those `PNGs`.

Comment (wmathurin) [reply iOSIDPTemplate/Images.xcassets/sam.imageset/Contents.json:5]: Not using the user’s profile pic?

Comment (bhariharan) [reply iOSIDPTemplate/Images.xcassets/sam.imageset/Contents.json:5]: Nope - we're not right now.

**Review by wmathurin — APPROVED** [review]

**Pull request merged**